src/v/raft/replicate_batcher.cc

```diff
@@ -72,14 +72,25 @@ replicate_batcher::cache_and_wait_for_result(
          * replicate batcher stop method
          *
          */
-        ssx::background
-          = ssx::spawn_with_gate_then(_bg, [this]() {
-                return _lock.get_units().then([this](auto units) {
-                    return flush(std::move(units), false);
-                });
-            }).handle_exception([item](const std::exception_ptr& e) {
-                item->set_exception(e);
+        if (!_flush_pending) {
+            _flush_pending = true;
+            ssx::background = ssx::spawn_with_gate_then(_bg, [this]() {
+                return _lock.get_units()
+                  .then([this](auto units) {
+                      return flush(std::move(units), false);
+                  })
+                  .handle_exception([this](const std::exception_ptr& e) {
```

Comment:
The handle_exception is moved from outside the spawn to inside since we use the consensus `ptr` which I think we need to do with the gate held.